test/data/base/03-include-in-define/example.yaml

```diff
@@ -0,0 +1,2 @@
+foo: "bar"
+bar: "foo"
```

Comment:
One more idea, inside the example we should do define a var based on a previous define, both in the simple and in the include case, something like:
```yaml
diff --git a/test/data/base/01-define.json b/test/data/base/01-define.json
index dd2ca6c..a42fb29 100644
--- a/test/data/base/01-define.json
+++ b/test/data/base/01-define.json
@@ -5,6 +5,9 @@
     },
     {
       "a": "tree/b"
+    },
+    {
+      "b": "tree"
     }
   ],
   "version": "2",
diff --git a/test/data/base/01-define.yaml b/test/data/base/01-define.yaml
index 12bb08d..f63e9e8 100644
--- a/test/data/base/01-define.yaml
+++ b/test/data/base/01-define.yaml
@@ -5,8 +5,10 @@ otk.version: "1"
 
 otk.define.name:
   a: "tree"
+  b: ${a}
 
 otk.target.osbuild.name:
   pipelines:
     - a: "${a}"
     - a: "${a}/b"
+    - b: "${b}"
diff --git a/test/data/base/03-include-in-define.json b/test/data/base/03-include-in-define.json
index d84def4..09d8985 100644
--- a/test/data/base/03-include-in-define.json
+++ b/test/data/base/03-include-in-define.json
@@ -2,7 +2,8 @@
   "pipelines": [
     {
       "foo": "bar",
-      "bar": "foo"
+      "bar": "foo",
+      "baz": "bar"
     }
   ],
   "version": "2",
diff --git a/test/data/base/03-include-in-define.yaml b/test/data/base/03-include-in-define.yaml
index f0cc68c..6a155b5 100644
--- a/test/data/base/03-include-in-define.yaml
+++ b/test/data/base/03-include-in-define.yaml
@@ -8,3 +8,4 @@ otk.target.osbuild.name:
   pipelines:
     - foo: ${subkey.foo}
       bar: ${subkey.bar}
+      baz: $subkey.baz}
diff --git a/test/data/base/03-include-in-define/example.yaml b/test/data/base/03-include-in-define/example.yaml
index bba4346..12c7125 100644
--- a/test/data/base/03-include-in-define/example.yaml
+++ b/test/data/base/03-include-in-define/example.yaml
@@ -1,2 +1,3 @@
 foo: "bar"
 bar: "foo"
+baz: ${foo}
```
(please double check). This will ensure that the parser "knows" when inside a define in an include, i.e. that it needs to keep track of newly includes vars inside a define as well. Maybe even add a dict here (baz:\n  subkey: something) so that we can be sure this part of the recursion is also working.

Reply:
I'm okay to merge this PR as is and we can PR this example along with other ones I have in mind.

Reply:
Sure, that works for me of course